examples/tests/valid/apps_with_params/nested_params/parameters.test.w

```diff
@@ -0,0 +1,24 @@
+struct Person {
+  name: str;
+  age: num;
+}
+
+struct House {
+  address: str;
+  residents: Array<Person>;
+}
+
+struct MyParams {
+  houses: Array<House>;
+}
+
+
+let app = nodeof(this).app;
+
+app.parameters.addSchema(MyParams.schema());
+
+let myParams = MyParams.fromParameters(app.parameters);
```

Comment:
`.fromParameters` feels a bit off. Not sure it makes sense to "pollute" all of our structs with a static method dedicated to reading `wing.toml` values. 

Maybe we can just use `fromJson`?

Maybe:

```js
MyParams.fromJson(app.parameters.json());
```

I am also wondering if the common case is to read all the parameters at once or portions of it... Feels like we need some kind of "sections" (e.g. `app.parameters.section("tf-aws")` which will return everything under the `[tf-aws]` section in the toml file...

Reply:
> I am also wondering if the common case is to read all the parameters at once or portions of it...

Im imagining the common case here is that it will be to read portions of it. Which is why the behavior of `.fromParameters()` will only return a subset of the parameters based on the Struct's schema. While I think `app.parameters.section("tf-aws")` will work for using `.fromJson` it also feels a bit strange to grab portions by string values. 

I dunno, Im not sure if I find it too odd to have `.fromParameters` since parameters are part of the core platform system. From a platform builder perspective I think it offers a pretty smooth DX. Though I do understand the concern with additional static methods.

Reply:
Though actually maybe that means that `.fromJson` should work the same way, that given an object will only extract the fields that match its schema... I think that might be a bug. We dont error on extraneous fields (which is intentional) but we should ignore them and not assign them to the struct.

Sorry for the wall of text Im just brain dumping 😅 Im running on 0 caffeine from Ramadan so thoughts are lagging

Reply:
Okay I fixed `fromJson` to not just return the whole JSON if the validation is successful. As for adding `fromSection` I dont think we need to, since if you want your Struct to be parsed from a certain section of the wing.toml you can do
```js
Struct.fromJson(app.parameters.json().get("tf-aws"));
```

lemme know what you think @eladb and ignore most of my ramblings before these last 2 comments.

Reply:
Yea im aligned on just using `fromJson` now, ive made the changes